### DISCUSSION TITLE
adding elements atoctupole, atdecapole, atdodecapole with corresponding Class Names

### DISCUSSION THREAD

**Discussion by simoneliuzzo** [atcollab/at#684] [General]
Why not adding to AT element creators as 
atoctupole, atdecapole, atdodecapole 
with corresponding Class Names and plotting with different colors in the atplot?

**Comment by lfarv** [comment]
Adding new element creators is easy. Adding new classes is more difficult. Classes are tested in various functions scattered around in the code, so modifying them requires a full review of the whole code. The situation is different in Matlab and python:

- it's a bit easier in python because of inheritance: all multipoles, existing and potentially new ones, inherit from Multipole, so a function ignoring a new multipole element class will fall in the `Multipole` case, which is a reasonable fallback solution,
- In Matlab it's more tricky: what to do in a "switch case…" statement where an element does not fall in foreseen cases: we don't know if it is a multipole or another new kind of element.

Another point is: where should we stop? python has an Octupole class, Matlab not. So one could reasonably think of adding it in Matlab, but do we need more? And will the limit be frozen for ever, or will we need to redo the whole review when someone wants to extend the list?

Where needed, there are already cases where the kind of multipole is deduced from the lowest non-zero order of PolynomA/B. Not very elegant, but works without any order limitation (see `atguessclass.m`).

Anyway, it's not really critical: whatever we may forget while extending the list of Classes will have only "cosmetic" consequences: the tracking, at the origin of all physical results, ignores the classes, it relies only on passmethods…
